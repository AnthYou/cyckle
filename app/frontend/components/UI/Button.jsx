import React from "react";
import classes from "./Button.module.scss";

const Button = (props) => {
  // type BtnColor = primary | secondary | alternate
  const { color, children, className, btnProps } = props;
  const btnClasses = `${classes.Button} ${classes[color]} ${className}`;

  return <button className={btnClasses} {...btnProps}>{children}</button>
}

export default Button;
