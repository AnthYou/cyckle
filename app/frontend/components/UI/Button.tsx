import React from "react";
import classes from "./Button.module.scss";

type ButtonColor = "primary" | "secondary" | "alternate";

interface ButtonProps {
  color: ButtonColor;
  className?: string;
  children?: React.ReactNode;
  onClick?: () => void;
}

const Button = ({
  color,
  className,
  children,
  onClick,
}: ButtonProps): JSX.Element => {
  let btnClasses = `${classes.Button} ${classes[color]}`;

  if (className) {
    btnClasses = `${classes.Button} ${classes[color]} ${className}`;
  }

  return (
    <button className={btnClasses} onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
