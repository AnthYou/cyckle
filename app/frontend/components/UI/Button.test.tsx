import React from "react";
import renderer from "react-test-renderer";
import { vi } from "vitest";

import Button from "./Button";

describe('<Button />', () => {
  it('should render correctly without providing a class', () => {
    const component = renderer.create(<Button color="primary" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render correctly providing a class', () => {
    const component = renderer.create(<Button color="primary" className="uppercase" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render correctly providing a class and an onClick property', () => {
    const component = renderer.create(<Button color="primary" className="uppercase" onClick={vi.fn} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
});
