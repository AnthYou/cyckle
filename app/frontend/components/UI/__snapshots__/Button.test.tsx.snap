// Vitest Snapshot v1

exports[`<Button /> > should render correctly providing a class 1`] = `
<button
  className="_Button_4ea4ec _primary_4ea4ec uppercase"
/>
`;

exports[`<Button /> > should render correctly providing a class and an onClick property 1`] = `
<button
  className="_Button_4ea4ec _primary_4ea4ec uppercase"
  onClick={[Function]}
/>
`;

exports[`<Button /> > should render correctly without providing a class 1`] = `
<button
  className="_Button_4ea4ec _primary_4ea4ec"
/>
`;
